packages/ui/test/working-groups/modals/LeaveRoleModal.test.tsx

```diff
@@ -0,0 +1,161 @@
+import { UseAccounts } from '../../../src/accounts/providers/accounts/provider'
+import { ApiContext } from '../../../src/common/providers/api/context'
+import { MembershipContext } from '../../../src/memberships/providers/membership/context'
+import { MyMemberships } from '../../../src/memberships/providers/membership/provider'
+import { seedMembers } from '../../../src/mocks/data'
+import { seedWorkingGroups } from '../../../src/mocks/data/mockWorkingGroups'
+import { LeaveRoleModal } from '../../../src/working-groups/modals/LeaveRoleModal/LeaveRoleModal'
+import { WorkerWithDetails } from '../../../src/working-groups/types'
+import { alice } from '../../_mocks/keyring'
+import { getMember } from '../../_mocks/members'
+import { MockKeyringProvider, MockQueryNodeProviders } from '../../_mocks/providers'
+import { setupMockServer } from '../../_mocks/server'
+import {
+  stubApi,
+  stubDefaultBalances,
+  stubTransaction,
+  stubTransactionFailure,
+  stubTransactionSuccess,
+} from '../../_mocks/transactions'
+
+const WORKER: WorkerWithDetails = {
+  membership: {
+    id: '',
+    controllerAccount: '',
+  },
+  group: {
+    name: 'forum',
+    id: '1',
+  },
+  isLeader: false,
+  rewardPerBlock: 100,
+  earnedTotal: 2000,
+  stake: 2000,
+  status: '',
+  id: '1',
+}
+
+describe('UI: LeaveRoleModal', () => {
+  const api = stubApi()
+
+  const useMyMemberships: MyMemberships = {
+    active: undefined,
+    members: [],
+    setActive: (member) => (useMyMemberships.active = member),
+    isLoading: false,
+  }
+
+  let useAccounts: UseAccounts
+  let transaction: any
+
+  const server = setupMockServer()
+
+  beforeAll(async () => {
+    await cryptoWaitReady()
+
+    useAccounts = {
+      hasAccounts: true,
+      allAccounts: [alice],
+    }
+  })
+
+  beforeEach(async () => {
+    seedMembers(server.server)
+    seedWorkingGroups(server.server)
+    useMyMemberships.setActive(getMember('alice'))
+    stubDefaultBalances(api)
+    transaction = stubTransaction(api, 'api.tx.forumWorkingGroup.leaveRole')
+  })
+
+  it('Renders the modal', async () => {
+    await renderModal()
+    expect(screen.findByText('Leaving a position?')).toBeDefined()
+    expect(screen.findByText('Please remember that this action is irreversible.')).toBeDefined()
+    expect(screen.getByRole('button', { name: 'Leave the position anyway' })).toBeDefined()
+  })
+
+  describe('Sign step', () => {
+    it("With member's controller account in keyring", async () => {
+      await renderWithValidSigner()
+      await fireEvent.click(screen.getByRole('button', { name: 'Leave the position anyway' }))
+      expect(
+        screen.findByText("The transaction can only be signed with the membership's controller account.")
+      ).toBeDefined()
+      expect(screen.getByRole('button', { name: 'Sign and leave role' })).not.toBeDisabled()
+    })
+
+    it("Without member's controller account in keyring", async () => {
```

Comment:
That shouldn't be possible. You should only be allowed to leave a role for the worker that you control.

This will only be possible if:

*   you have the account key pair (from the extension)
*   the app loads a member for your keys
*   the app detects if that Worker is associated with a Member that is yours